app/controllers/conversions_controller.rb

```diff
@@ -14,14 +14,15 @@ def convert
     Rails.logger.info "converting shapefile #{request.headers["Content-Type"]}"
 
     # Define local_path method required by Faraday
-    upload = params[:upload]
+    upload = safe_hash(:upload)
     def upload.local_path
       @tempfile
     end
 
-    response = OgreClient.convert_shapefile(params)
+    response = OgreClient.convert_shapefile(params.permit!.to_h)
+    
 
     # render(json: current_user.to_fileupload(name, style), content_type: request.format)
-    render text: response.body, content_type: request.format
+    render :json => response.body
```

Comment:
I think render json: response.body is the preferred syntax.

Reply:
This produces a syntax error:
SyntaxError
/Users/jdfarley/Work/mmt/app/controllers/conversions_controller.rb:26: syntax error, unexpected tIDENTIFIER, expecting keyword_end
    render :json response.body
                 ^~~~~~~~
Extracted source (around line #26):

It took several tries to find "render :json => response.body" worked correctly.   render has gone through some changes in Rails 5

Thoughts?

Reply:
Does 'json:' cause a syntax error?  json: response.body should be interpreted as a hash here.

Reply:
Sorry,  mis-typed!  render JSON: response.body  does work.  I'll incorporate this change.